src/components/EntitlementsPanelForm/Fields/Fields.tsx

```diff
@@ -0,0 +1,122 @@
+import { Select } from "@canonical/react-components";
+import { useQueryClient } from "@tanstack/react-query";
+import { useFormikContext } from "formik";
+
+import type { EntitlementSchema, EntityEntitlement } from "api/api.schemas";
+import { useGetResources } from "api/resources/resources";
+import CleanFormikField from "components/CleanFormikField";
+import FormikSubmitButton from "components/FormikSubmitButton";
+import { Endpoint } from "types/api";
+
+import { EntitlementsPanelFormLabel } from "..";
+
+import { Label } from "./types";
+
+type Props = {
+  entitlements: EntitlementSchema[];
+};
+
+const Fields = ({ entitlements }: Props) => {
+  const queryClient = useQueryClient();
+  const { values, setFieldValue, setFieldTouched } =
+    useFormikContext<EntityEntitlement>();
+  const { data: getResourcesData, isFetching: isGetResourcesFetching } =
+    useGetResources(
```

Comment:
I think that we should probably leave a TODO here noting that this will only fetch the first page of results.